plugin.js

```diff
@@ -63,6 +63,7 @@ function onReqHandlerWrapper (fastify) {
 
 function plugin (fastify, options, next) {
   const secret = options.secret
+  const addHookOnRequest = options.addHookOnRequest ?? true
```

Comment:
can we rename the flag? It describes what we do in plugin, but if you consume the plugin in your implementation, you are forced to check the sourcecode/readme of fastify/cookie to understand what it means. 

it should be something like parseCookieOnRequest or autoParseCookieOnRequest.

Reply:
I would prefer we add a `hook` option instead that default to `onRequest` but that it can also be `false`. If `false` we are not setting the hook at all.

Reply:
yes, of course you can rename the flag. Let's think and choose some option:
 - parseCookieOnRequest
 - autoParseCookieOnRequest
 - autoParseCookie

Reply:
see my comment